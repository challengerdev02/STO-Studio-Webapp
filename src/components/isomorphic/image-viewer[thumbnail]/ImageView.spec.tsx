import {
  cleanup,
  fireEvent,
  queryAllByAltText,
  queryAllByRole,
  queryByAttribute,
  queryByRole,
  render,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ImageViewComponent } from './index';

describe('ImageViewComponent', function () {
  afterEach(cleanup);
  let props = {
    images: [
      {
        src: 'https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk',
      },
      {
        src: 'https://i.picsum.photos/id/848/600/400.jpg?hmac=gdx1i308nOCrjtfzwAlWF5RCN8HKoQd-POMetrAmZxc',
      },
    ],
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<ImageViewComponent {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should display the main image container', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryAllByRole(document.body, 'main-image-view')[0]
    ).toBeInTheDocument();
    expect(queryAllByRole(document.body, 'main-image-view')[0]).toHaveClass(
      'main-image-view'
    );
  });

  it('should display the thumbnail container', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryByRole(document.body, 'thumbnail-container')
    ).toBeInTheDocument();
  });

  it('should have Images', () => {
    // @ts-ignore
    const {} = setup();
    expect(queryAllByAltText(document.body, 'image')[0]).toBeInTheDocument();
    expect(queryAllByAltText(document.body, 'image')[0]).toHaveAttribute(
      'src',
      'https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk'
    );
  });

  it('should have a expand button', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryByAttribute('class', document.body, 'ant-image-mask')
    ).toBeInTheDocument();
  });

  it('should display with NO image container', () => {
    // @ts-ignore
    const { container } = setup({ images: [] });
    expect(queryAllByRole(document.body, 'main-image-view').length).toBe(0);
  });

  it('should fire onCancel callback when modal is closed', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryByAttribute('class', document.body, 'ant-image-mask');

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'dialog')).toBeInTheDocument();
      expect(queryByRole(document.body, 'dialog')).not.toHaveStyle(
        'display: none'
      );
    });
  });

  it('should fire Next callback Next Thumbnail is clicked', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryAllByRole(document.body, 'unit-thumbnail')[1];
      expect(btn).toBeInTheDocument();
      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      const btn = queryAllByRole(document.body, 'unit-thumbnail')[1];
      const active = queryByAttribute(
        'class',
        btn,
        'ant-image-img thumbnail-image-view active-thumbnail'
      );
      expect(active).toBeInTheDocument();
    });
  });
});
