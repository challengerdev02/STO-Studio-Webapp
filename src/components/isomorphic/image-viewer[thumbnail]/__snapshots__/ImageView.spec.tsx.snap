// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewComponent should match with snapshot 1`] = `
<div>
  <div
    class="sc-gsDKAQ gwlFDG image-view-container"
  >
    <div
      class="sc-bdvvtL oKzls image-view-container main-image-view"
      role="main-image-view"
    >
      <div
        class="ant-image"
        role="main-image-view"
      >
        <img
          alt="image"
          class="ant-image-img main-image-view"
          src="https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
    </div>
    <div
      style="display: none;"
    >
      <div
        class="ant-image"
      >
        <img
          alt="...."
          class="ant-image-img"
          src="https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
      <div
        class="ant-image"
      >
        <img
          alt="...."
          class="ant-image-img"
          src="https://i.picsum.photos/id/848/600/400.jpg?hmac=gdx1i308nOCrjtfzwAlWF5RCN8HKoQd-POMetrAmZxc"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-dkPtRN gWvrvz thumbnail-container"
      role="thumbnail-container"
    >
      <div
        class="sc-bdvvtL oKzls image-view-container thumbnail-image-view active-thumbnail"
        role="unit-thumbnail"
      >
        <div
          class="ant-image"
          nopreview="true"
          role="unit-thumbnail"
        >
          <img
            alt="image"
            class="ant-image-img thumbnail-image-view active-thumbnail"
            src="https://i.picsum.photos/id/111/600/400.jpg?hmac=hVzgLIvxNZYaZkGV8HgOs6FO0aePt5tSEhb-sZBwbkk"
          />
        </div>
      </div>
      <div
        class="sc-bdvvtL oKzls image-view-container thumbnail-image-view "
        role="unit-thumbnail"
      >
        <div
          class="ant-image"
          nopreview="true"
          role="unit-thumbnail"
        >
          <img
            alt="image"
            class="ant-image-img thumbnail-image-view "
            src="https://i.picsum.photos/id/848/600/400.jpg?hmac=gdx1i308nOCrjtfzwAlWF5RCN8HKoQd-POMetrAmZxc"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
